### PR TITLE
RPM 4.15 support

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

**Comment by matejak** [comment]
add to whitelist

File: CMakeLists.txt [hunk 1]
```diff
@@ -132,6 +132,7 @@ if(RPM_FOUND)
 	check_library_exists("${RPM_LIBRARY}" headerFormat "" HAVE_HEADERFORMAT)
 	check_library_exists("${RPMIO_LIBRARY}" rpmFreeCrypto "" HAVE_RPMFREECRYPTO)
 	check_library_exists("${RPM_LIBRARY}" rpmFreeFilesystems "" HAVE_RPMFREEFILESYSTEMS)
+	check_library_exists("${RPM_LIBRARY}" rpmVerifyFile "" HAVE_RPMVERIFYFILE)
```
Comment (jan-cerny) [inline CMakeLists.txt]: There is a warning on RHEL7:
```
/home/jenkins/workspace/openscap-pull-requests-multi/label/rhel7/src/OVAL/probes/unix/linux/rpm-helper.c:39:5: warning: implicit declaration of function ‘rpmfiVerify’ [-Wimplicit-function-declaration]
     rpmVerifyAttrs vfy = rpmfiVerify(fi, omitMask);
     ^
```

and error:

```
Linking C executable oscap
../src/libopenscap.so.25.0.0: undefined reference to `rpmfiVerify'
collect2: error: ld returned 1 exit status
gmake[2]: *** [utils/oscap] Error 1
```


I think you need to add `#cmakedefine HAVE_RPMVERIFYFILE` to `config.h.in` around line 42 to fix this.

File: src/OVAL/probes/unix/linux/rpmverify_probe.c [hunk 2]
```diff
@@ -303,19 +303,14 @@ typedef struct {
 } rpmverify_bhmap_t;
 
 const rpmverify_bhmap_t rpmverify_bhmap[] = {
-        { "nodeps",        (uint64_t)VERIFY_DEPS      },
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/linux/rpmverify_probe.c]: Is there any way how to keep those items with rpmlib >= 4.15? These are defined by the [OVAL spefification](https://oval.mitre.org/language/version5.11/ovaldefinition/documentation/linux-definitions-schema.html#RpmVerifyBehaviors). They are marked as deprecated, but we should still be able to process them.

**Comment by matejak** [comment]
test this please

File: src/OVAL/probes/unix/linux/rpmverify_probe.c (hunk 2 above)
Comment (jan-cerny) [reply src/OVAL/probes/unix/linux/rpmverify_probe.c]: Hi @ignatenkobrain , do you have any information about this?

**Comment by jan-cerny** [comment]
@openscap-ci test this please

File: src/OVAL/probes/unix/linux/rpmverify_probe.c (hunk 2 above)
Comment (evgenyz) [reply src/OVAL/probes/unix/linux/rpmverify_probe.c]: VERIFY_DEPS was never a part of flags for rpmVerifyFile. This is a flag for rpmcliVerify, which is not used in this probe at all. Whatever this probe was supposed to do regarding the --nodeps flag it never did.

**Pull request merged**